segmentstore/server/src/main/java/io/pravega/segmentstore/server/writer/StorageWriter.java

```diff
@@ -513,7 +528,8 @@ private Duration getReadTimeout() {
      */
     private Duration getIterationStartDelay() {
         if (this.state.getLastIterationError()) {
-            return this.config.getErrorSleepDuration();
+            // Simple linear backoff.
+            return Duration.ofMillis(this.state.getFailedIterationCount() * this.config.getErrorSleepDuration().toMillis());
```

Comment:
Retry is implemented and used differently - it takes a lambda to execute which it tries a few times.
Instead here we are already in a "for loop" continuously running iterations until asked to stop.
And we dynamically need to adjust delay based on past iterations.

Nothing is impossible, It is possible to use Retry in some hacky way.
But here situation is rather simple so just using simple approach of directly calculating delay.
I agree there should be max delay and it can't be unbounded. I'll make that change